aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1112,6 +1112,15 @@ public <T extends S3ClientBuilder> void applyS3SignerConfiguration(T builder) {
     }
   }
 
+  /**
+   * Get the AWS Credentials Provider based on the configured credential properties.
+   *
+   * @return credential provider
+   */
+  AwsCredentialsProvider awsCredentialsProvider() {
+    return credentialsProvider(s3AccessKeyId, s3SecretAccessKey, s3SessionToken);
```

Comment:
I am thinking should we treat the signer more like an AWS client and have its own set of credential properties? It does not really have association with S3 access credentials and it does not make much sense to reuse those property values.

Reply:
I pulled them out of AwsProperties and use different property names.  Let me know if this is closer to what you had in mind.